config/network.go

```diff
@@ -0,0 +1,161 @@
+package config
+
+import (
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"errors"
+
+	"github.com/pelletier/go-toml/v2"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/osutil"
+)
+
+const (
+	Base64NetworkConfigEnvVarName = "BASE64_NETWORK_CONFIG"
+)
+
+type NetworkConfig struct {
+	SelectedNetworks []string            `toml:"selected_networks"`
+	RpcHttpUrls      map[string][]string `toml:"RpcHttpUrls"`
+	RpcWsUrls        map[string][]string `toml:"RpcWsUrls"`
+	WalletKeys       map[string][]string `toml:"WalletKeys"`
+}
+
+func (n *NetworkConfig) ApplySecrets() error {
+	encodedEndpoints, err := osutil.GetEnv(Base64NetworkConfigEnvVarName)
+	if err != nil {
+		return err
+	}
+
+	err = n.ApplyBase64Enconded(encodedEndpoints)
+	if err != nil {
+		return fmt.Errorf("error reading network encoded endpoints: %w", err)
+	}
+
+	return nil
+}
+
+func (n *NetworkConfig) ApplyDecoded(configDecoded string) error {
+	if configDecoded == "" {
+		return nil
+	}
+
+	var cfg NetworkConfig
+	err := toml.Unmarshal([]byte(configDecoded), &cfg)
+	if err != nil {
+		return fmt.Errorf("error unmarshaling network config: %w", err)
+	}
+
+	err = n.ApplyOverrides(&cfg)
+	if err != nil {
+		return fmt.Errorf("error applying overrides from decoded network config file to config: %w", err)
+	}
+
+	return nil
+}
+
+func (n *NetworkConfig) ApplyBase64Enconded(configEncoded string) error {
+	if configEncoded == "" {
+		return nil
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(configEncoded)
+	if err != nil {
+		return err
+	}
+
+	return n.ApplyDecoded(string(decoded))
+}
+
+func (n *NetworkConfig) Validate() error {
+	if len(n.SelectedNetworks) == 0 {
+		return errors.New("selected_networks must be set")
+	}
+
+	upperCasedHttpUrls := make(map[string][]string)
+	for key := range n.RpcHttpUrls {
+		upperCasedHttpUrls[strings.ToUpper(key)] = n.RpcHttpUrls[key]
+	}
+	n.RpcHttpUrls = upperCasedHttpUrls
+
+	upperCasedWsUrls := make(map[string][]string)
+	for key := range n.RpcWsUrls {
+		upperCasedWsUrls[strings.ToUpper(key)] = n.RpcWsUrls[key]
+	}
+	n.RpcWsUrls = upperCasedWsUrls
+
+	upperCasedWalletKeys := make(map[string][]string)
+	for key := range n.WalletKeys {
+		upperCasedWalletKeys[strings.ToUpper(key)] = n.WalletKeys[key]
+	}
+	n.WalletKeys = upperCasedWalletKeys
+
+	for i, network := range n.SelectedNetworks {
+		n.SelectedNetworks[i] = strings.ToUpper(network)
+	}
+
+	for _, network := range n.SelectedNetworks {
+		if strings.Contains(network, "SIMULATED") {
+			// we don't need to validate RPC endpoints or private keys for simulated networks
+			continue
+		}
+
+		if _, ok := n.RpcHttpUrls[network]; !ok {
+			return fmt.Errorf("At least one HTTP RPC endpoint for %s network must be set", network)
+		}
+
+		if _, ok := n.RpcWsUrls[network]; !ok {
+			return fmt.Errorf("At least one WS RPC endpoint for %s network must be set", network)
+		}
+
+		if _, ok := n.WalletKeys[network]; !ok {
```

Comment:
how will we deal with keys which we don't want to know and want them to be managed by security? cc @kalverra

Reply:
could you expand on that?

what I'm doing now in live testnet job is this:
* that job takes EVM keys from GH secrets (so GH automatically masks it in log output)
* I build base64 config override with network details, including the private kay and mask it with `::add-mask::`
* I set that override as env var that tests read

Do you see any potential risk with that approach?

Reply:
My question was whether we could use EVM Keys from github secrets, if we can do that, my question is answered.